docs/how-to-guides/performance_analysis.md

```diff
@@ -0,0 +1,206 @@
+# Performance analysis
+
+## Introduction
+
+Autoware is a real-time system, and it is important to have a small response time. If Autoware appears to be slow, it is imperative to conduct performance measurements and implement improvements based on the analysis. However, Autoware is a complex software system comprising numerous ROS 2 nodes, potentially complicating the process of identifying bottlenecks. To address this challenge, we will discuss methods for conducting detailed performance measurements for Autoware and provide case studies. It is worth noting that multiple factors can contribute to poor performance, such as scheduling and memory allocation in the OS layer, but our focus in this page will be on user code bottlenecks. The outline of this section is as follows:
```

Comment:
It might be misleading to say "Autoware is a real-time system" because it currently is not. It even depends on user's usecase whether it should be real-time system.

Maybe something like the following might be better:
"When integrating Autoware to an actual service, user may want to set a requirement on response time. If Autoware appears to be slow ..."

Reply:
Being a real-time system means that the correctness of the output includes a time constraint.
So Autoware is a real-time system.

Reply:
I understand what real-time system is.
What I am saying is that Autoware is not currently real-time as is. There is no requirements about time constraint defined anywhere in Autoware because it depends on users definition, and there is no code in Autoware that ensures that operation finishes in given period of time. 
"Autoware should be real-time system" or "Autoware integrated-system should be real-time system" might be correct statements (in my opinion this also depends on usecase), but "Autoware is real-time system" is not true.